src/sentry/lang/native/appconnect.py

```diff
@@ -93,18 +102,10 @@ class AppStoreConnectConfig:
     # This is guaranteed to be unique and should map 1:1 to ``appId``.
     bundleId: str
 
-    # TODO(itunes): Deprecated fields. These must be removed alongside a migration.
-    itunesUser: str
-    itunesPassword: str
-    itunesSession: str
-    itunesCreated: datetime
-    orgPublicId: str
-    orgName: str
-
     def __post_init__(self) -> None:
-        # All fields are required.
         for field in dataclasses.fields(self):
-            if not getattr(self, field.name, None):
+            # Takes advantage of the fact that all non-optional fields are str.
+            if not getattr(self, field.name, None) and field.type == str:
```

Comment:
I don't really understand why this condition needed changing?  You removed the fields, all fields are still required.

I don't think this does anything wrong either, this will still work.  I only don't know why the extra `field.type == str` check is needed.